src/cheri_sys_regs.sail

```diff
@@ -32,11 +43,11 @@ function legalize_ccsr(c : ccsr, v : xlenbits) -> ccsr = {
 
 function clause ext_read_CSR (0x8C0) = Some(uccsr.bits())
 function clause ext_read_CSR (0x9C0) = Some(sccsr.bits())
-function clause ext_read_CSR (0xBC0) = Some(mccsr.bits())
+function clause ext_read_CSR (0xBC0) = Some(sccsr.bits())
 
-function clause ext_write_CSR (0x8C0, value) = { uccsr = legalize_ccsr(uccsr, value); Some(uccsr.bits()) }
-function clause ext_write_CSR (0x9C0, value) = { sccsr = legalize_ccsr(sccsr, value); Some(sccsr.bits()) }
-function clause ext_write_CSR (0xBC0, value) = { mccsr = legalize_ccsr(mccsr, value); Some(mccsr.bits()) }
+function clause ext_write_CSR (0x8C0, value) = { uccsr = legalize_uccsr(uccsr, value); Some(uccsr.bits()) }
+function clause ext_write_CSR (0x9C0, value) = { sccsr = legalize_sccsr(sccsr, value); Some(sccsr.bits()) }
+function clause ext_write_CSR (0xBC0, value) = { sccsr = legalize_sccsr(sccsr, value); Some(sccsr.bits()) }
```

Comment:
Maybe nicer to pass in Machine/Supervisor/User rather than duplicating the function?